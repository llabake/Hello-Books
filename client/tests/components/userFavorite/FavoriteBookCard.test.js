import React from 'react';

import {
  FavoriteBookCard,
  mapDispatchToProps,
  mapStateToProps
} from '../../../components/userFavorite/FavoriteBookCard';
import adminData from '../../mocks/userData';

function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    handleDelete: jest.fn(),
    removeFromFavorite: jest.fn().mockImplementation(() => Promise.resolve(1)),
    handleSelectedPage: jest.fn(),
    errors: {},
    favoriteBook: {
      "createdAt": "2018-03-14T07:03:45.278Z",
      "book":
        {
          "id": 3,
          "title": "Never eat alone ",
          "description": "Finally we have a real-world guide to how to create your own high-powered network tailored to your career goals and personal style.\"",
          "upVotes": 2,
          "downVotes": 1,
          "image": "book image",
          "reviews": [],
          "favorited": []
        }
    }
  }

  const wrapper = shallow(<FavoriteBookCard {...props} />);

  return {
    wrapper,
    props,
  };
};


describe('Favorite Card', () => {
  test('should render correctely', () => {
    const { wrapper, props } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('should handle favorite book delete action', () => {
    const { wrapper, props } = setup();
    const event = {
      preventDefault: jest.fn(),
      target: {}
    }
    props.removeFromFavorite.mockReset()
    wrapper.instance().handleDelete(event)
    // expect(props.removeFromFavorite).toHaveBeenCalled()
    // expect(props.handleDelete).toHaveBeenCalled()
  });

  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('removeFromFavorite');

    const { removeFromFavorite } = mapDispatchToProps(dispatch);
    removeFromFavorite();
    expect(dispatch).toHaveBeenCalled();
  });


  test('mapStateToProps', () => {
    expect(mapStateToProps({ userReducer: {} })).toHaveProperty('errors');
  });
})