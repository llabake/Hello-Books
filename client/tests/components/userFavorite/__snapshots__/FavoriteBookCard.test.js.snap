// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Card should render correctely 1`] = `
<div
  className="col s12 m6 l4"
>
  <div
    className="card favorite-card"
  >
    <div
      className="card-image"
      style={
        Object {
          "maxHeight": "175px",
          "minHeight": "150px",
          "paddingTop": "10px",
        }
      }
      title="Never eat alone "
    >
      <Link
        replace={false}
        to="/book/3"
      >
        <img
          alt="Book Image"
          src="book image"
          style={
            Object {
              "maxHeight": "250px",
            }
          }
        />
      </Link>
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingTop": "7em",
        }
      }
    >
      <span
        className="card-title truncate"
        title="Never eat alone "
      >
        Never eat alone 
      </span>
      <p
        className="truncate"
        style={
          Object {
            "height": "1em",
            "minHeight": "2em",
          }
        }
        title="Finally we have a real-world guide to how to create your own high-powered network tailored to your career goals and personal style.\\""
      >
        Finally we have a real-world guide to how to create your own high-powered network tailored to your career goals and personal style."
      </p>
      <br />
      <Link
        className="read-more right"
        replace={false}
        to="/book/3"
      >
        Read More
      </Link>
    </div>
    <div
      className="card-action"
    >
      <ul>
        <li>
          <a
            style={
              Object {
                "marginRight": "0px",
              }
            }
          >
            <i
              className=" material-icons fav-icon"
            >
              favorite
            </i>
          </a>
          <span>
             
            <strong>
              0
            </strong>
          </span>
        </li>
        <li>
          <a
            style={
              Object {
                "marginLeft": "1.5em",
                "marginRight": "0px",
              }
            }
          >
            <i
              className="small material-icons up-icon"
            >
              thumb_up
            </i>
          </a>
          <span>
             
            <strong>
              2
            </strong>
          </span>
        </li>
        <li>
          <a
            style={
              Object {
                "marginLeft": "1.5em",
                "marginRight": "0px",
              }
            }
          >
            <i
              className="small material-icons down-icon"
            >
              thumb_down
            </i>
          </a>
          <span>
              
            <strong>
              1
            </strong>
          </span>
        </li>
        <li>
          <a
            style={
              Object {
                "marginLeft": "1.5em",
                "marginRight": "0px",
              }
            }
          >
            <i
              className="small material-icons comment-icon"
            >
              comment
            </i>
          </a>
          <span>
              
            <strong>
              0
            </strong>
          </span>
        </li>
      </ul>
    </div>
    <div
      className="card-content"
      style={
        Object {
          "cursor": "pointer",
          "paddingTop": "0px",
        }
      }
    >
      <span
        id="delete-favorite"
      >
        <a
          onClick={[Function]}
        >
          Remove
        </a>
      </span>
    </div>
  </div>
</div>
`;
